server/utils/passport/local-signup.js

```diff
@@ -1,96 +1,89 @@
-            acl: 1,
-            role: 1,
-            icon: 1,
-            mail: 1
+      try {
+        const password = reqBody.password
+        const username = reqBody.username
+        const mail = reqBody.email
+        const display_name = reqBody.display_name
+        const hashedPW = hash(password)
+        const newUserData = {
+          uid: username,
+          display_name,
+          title: '',
+          department: '',
+          company: '',
+          mail,
+          member_of: '',
+          bad_pwd_count: 0,
+          lockout_time: 0,
+          last_logoff: Date.now(),
+          last_logon: Date.now(),
+          password: hashedPW,
+          ldap: false,
+          force_reset_pw: false,
+          realms: config.app.realms,
+          icon: '',
+          access: [],
+          blocked: false,
+          role: 'member',
+          preferences: {
+            config: '',
+            sort: 0,
+            star: [],
+            complete: [],
           },
-          returnOriginal: false,
-          upsert: true
-        },
-        function (err, userinfo) {
-          if (err) {
-            console.error(err.message);
-          }
+        }
+        const { prisma, appDb } = req.app.locals
+        const newUser = await prisma.users.create({
+          data: newUserData,
+          select: {
+            uid: true,
+            display_name: true,
+            access: true,
+            role: true,
+            icon: true,
+            mail: true,
+          },
+        })
+        const defaultConfig = {
+          owner: username,
+          config: defaultUserConfig,
+          name: 'default',
+          type: 'matrix',
+          readers: [username],
+          created: new Date().toUTCString(),
+        }
+        await appDb.collection(collections.configs).insertOne(defaultConfig)
 
-          var defaultConfig = {
-            owner: username,
-            config: defaultUserConfig,
-            name: 'default',
-            type: 'matrix',
-            readers: [username],
-            created: (new Date()).toUTCString()
-          };
-          mongodb.collection('configs').insertOne(defaultConfig);
-          client.close();
+        req.session.role = newUser.role
+        req.session.uid = newUser.uid
+        req.session.display_name = newUser.display_name
+        req.session.mail = newUser.mail
+        req.session.celery_tasks = []
+        req.session.icon = newUser.icon
 
-          req.session.role = userinfo.value.role;
-          req.session.uid = userinfo.value.uid;
-          req.session.display_name = userinfo.value.display_name;
-          req.session.mail = userinfo.value.mail;
-          req.session.celery_tasks = [];
-          req.session.icon = userinfo.value.icon;
-          return res.redirect(`${basePath}/`);
-        });
-    });
-  })(req, res, next);
-};
+        return res.redirect(`${basePath}/`)
+      } catch (error) {
+        console.log(error, 'THIS IS THE ERror')
```

Comment:
did you mean to leave the end string here?

Reply:
https://github.com/AMP-SCZ/dpdash/pull/241/commits/2e3f2e5a20221046bc208d7918023e42684da6e2

Reply:
That commit does not resolve this comment.